src/lib.rs

```diff
@@ -75,16 +76,18 @@ pub enum HostType {
 
 impl From<c_uint> for HostType {
     fn from(value: c_uint) -> Self {
-        match value {
-            0 => Self::Domain,
-            1 => Self::IPV4,
-            2 => Self::IPV6,
-            _ => Self::Domain,
+        if value > 2 {
+            return Self::Domain;
         }
+
+        // Safety: Prior to transmuting we checked if value is bigger than 2
```

Comment:
Can you add the mapping (number to enum) as a comment to here? Let's not also remove the knowledge we have since it might not be obvious at the first glance.